.github/workflows/ci.yaml

```diff
@@ -0,0 +1,40 @@
+name: ci-workflow
+# run workflow for these events
+on: ['push', 'pull_request', 'workflow_dispatch']
+
+jobs:
+
+  build:
+    strategy:
+      matrix:
+        python-version: ['3.6', '3.7', '3.8', '3.9']
+        os: ['ubuntu-latest', 'macos-latest']
+      fail-fast: false
+
+    name: Python ${{ matrix.python-version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: pip install --upgrade --requirement requirements-ci.txt
+
+      - name: Cache nltk data
+        uses: actions/cache@v2
+        with:
+          path: ~/nltk_data
+          key: nltk_data
+
+      - name: Download nltk data packages
+        run: python -c "import nltk; nltk.download('all')"
+        shell: bash
+
+      - name: Run pytest
+        run: pytest
```

Comment:
can you change this to `pytest nltk/test`? for some reason i don't totally understand, running this way doesn't catch the doctests

Reply:
Good call

Running `pytest` will use a different root directory for running tests will not include configuration from `pytest.ini` which is what allows doctests to run when using `pytest nltk/test`. I originally was going to leave doctests as an item for future work but have now implemented running doctests in the current version of this PR.

Also added caching for python dependencies and added parallel execution for tests.

Reply:
So awesome! Great job

Reply:
Thanks! :heart: